### ISSUE TITLE
Exception while using BetweennessCentrality algorithm 

### ISSUE THREAD

**Issue by Toptachamann** [jgrapht/jgrapht#734]
Hi, 

I'm in the process of deprecating the old version of FibonacciHeap. When I try to benchmark the BetweennessCentrality algorithm on the following directed weighted graph:
```
4 6
1 0 5
2 0 1
2 1 9
3 0 9
3 1 2
3 2 7
```
, I get the following exception:

``` console
Exception in thread "main" java.lang.IllegalArgumentException: decreaseKey() got larger key value. Current key: 7.0 new key: 8.0
	at org.jgrapht.util.FibonacciHeap.decreaseKey(FibonacciHeap.java:112)
	at org.jgrapht.alg.scoring.BetweennessCentrality$WeightedQueue.update(BetweennessCentrality.java:247)
	at org.jgrapht.alg.scoring.BetweennessCentrality$WeightedQueue.update(BetweennessCentrality.java:225)
	at org.jgrapht.alg.scoring.BetweennessCentrality.compute(BetweennessCentrality.java:190)
	at java.util.LinkedHashMap$LinkedKeySet.forEach(LinkedHashMap.java:559)
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1080)
	at org.jgrapht.alg.scoring.BetweennessCentrality.compute(BetweennessCentrality.java:133)
	at org.jgrapht.alg.scoring.BetweennessCentrality.getScores(BetweennessCentrality.java:103)
	at org.jgrapht.alg.matching.blossom.v5.MyGenerator.main(MyGenerator.java:31)
```

---------------------------------

**Edit:** there are graphs, on which this algorithm works without exceptions

**Comment by d-michail** [comment]
The problem seems to be that the following code 
```
                // shortest path to w via v?
                if (distance.get(w) >= d) {
                    queue.update(w, d);
                    sigma.put(w, sigma.get(w) + sigma.get(v));
                    predecessors.get(w).add(v);
                }
```
does not update the distance array.

**Comment by d-michail** [comment]
@Toptachamann Could you add a `distance.put(w, d);` inside the if statement above in #735 ?

**Comment by d-michail** [comment]
Fixed in PR #735.
